tests/test_synced_collection.py

```diff
@@ -345,14 +345,24 @@ def test_keys_with_dots(self, synced_dict):
         with pytest.raises(InvalidKeyError):
             synced_dict['a.b'] = None
 
-    def test_keys_valid_type(self, synced_dict, testdata):
+    def test_keys_str_type(self, synced_dict, testdata):
 
         class MyStr(str):
             pass
-        for key in ('key', MyStr('key'), 0, None, True):
+        for key in ('key', MyStr('key')):
             synced_dict[key] = testdata
-            assert str(key) in synced_dict
-            assert synced_dict[str(key)] == testdata
+            assert key in synced_dict
+            assert synced_dict[key] == testdata
+
+    # The following test tests the support for non-str keys
+    # for JSON backend which will be removed in version 2.0.
+    # see issue #316.
+    def test_keys_non_str_valid_type(self, synced_dict, testdata):
+        if synced_dict.backend == 'signac.core.collection_json':
```

Comment:
Why does the `synced_dict.backend` property exist? A type check seems more appropriate, especially since `JSONDict` has already been imported.
```suggestion
        if isinstance(synced_dict, JSONDict):
```

Reply:
Upon further investigation, I'm almost positive that `backend` should be a private property, and it may be possible to eliminate it altogether. I think you can handle the registry using the knowledge you have about synced collection classes without needing to define a string for the backend name.

Reply:
`synced_dict.backend` is not a property. It is a class attribute. Will change it to private. But I don't think we can remove this. This also used to find type of child class (JSONDict child should be JSONDict and JSONList ).